x-pack/plugins/apm/server/lib/anomaly_detection/create_anomaly_detection_jobs.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Logger } from 'kibana/server';
+// import uuid from 'uuid/v4';
+// import { Job as AnomalyDetectionJob } from '../../../../ml/server';
+import { PromiseReturnType } from '../../../../observability/typings/common';
+import { Setup } from '../helpers/setup_request';
+
+export type CreateAnomalyDetectionJobsAPIResponse = PromiseReturnType<
+  typeof createAnomalyDetectionJobs
+>;
+export async function createAnomalyDetectionJobs(
+  setup: Setup,
+  environments: string[],
+  logger: Logger
+) {
+  const { ml } = setup;
+  if (!ml) {
+    return [];
+  }
+  const mlCapabilities = await ml.mlSystem.mlCapabilities();
+  if (
+    !(
+      mlCapabilities.mlFeatureEnabledInSpace &&
+      mlCapabilities.isPlatinumOrTrialLicense
+    )
+  ) {
+    logger.warn('Anomaly detection integration is not availble for this user.');
+    return [];
+  }
+  logger.info(
+    `Creating ML anomaly detection jobs for environments: [${environments}]...`
+  );
+
+  const result = await Promise.all(
+    environments.map((environment) => {
+      return configureAnomalyDetectionJob({ ml, environment });
+    })
+  );
+  return result;
+}
+
+async function configureAnomalyDetectionJob({
+  ml,
+  environment,
+}: {
+  ml: Required<Setup>['ml'];
+  environment: string;
+}) {
+  const convertedEnvironment = environment; // TODO convert to ML id compatible format
+  const randomToken = '1234'; // TODO use generated UUID
+  const moduleId = 'apm_transaction';
+  const prefix = `apm-${convertedEnvironment}-${randomToken}-`;
+  const groups = ['apm', convertedEnvironment];
+  const indexPatternName = 'apm-*-transaction-*';
+  const query = {
+    bool: {
+      filter: [
+        { term: { 'processor.event': 'transaction' } },
+        { exists: { field: 'transaction.duration.us' } },
+        { term: { 'service.environment': environment } },
+      ],
+    },
+  };
+  const useDedicatedIndex = false;
+  const startDatafeed = true;
+  const start = Date.now(); // TODO what should these start/end values be?
+  const end = Date.now();
+  const jobOverrides = [
+    {
+      custom_settings: {
+        job_tags: {
+          'service.environment': environment,
+        },
+      },
+    },
+  ];
+  const datafeedOverrides: any[] = []; // TODO is this required?
+
+  return ml.modules.saveModuleItems(
+    moduleId,
+    prefix,
+    groups,
+    indexPatternName,
+    query,
+    useDedicatedIndex,
+    startDatafeed,
+    start,
+    end,
+    jobOverrides, // Typescript Error: '{ job_tags: { 'service.environment': string; }; }' has no properties in common with type 'CustomSettings'.
```

Comment:
`job_tags` will need to be added here:
https://github.com/elastic/kibana/blob/83beede50cb57f012411cc31952692d8cd888d52/x-pack/plugins/ml/common/types/anomaly_detection_jobs/job.ts#L13-L16

Reply:
Great, thanks for this @jgowdyelastic.

Currently `datafeedOverrides`, `start`, and `end` are required in `saveModuleItems`. They were not required on the `POST /api/ml/modules/setup/apm_transaction` endpoint. Can we do something about this?

Reply:
it looks like with the latest version these args types are unioned with `| undefined`, so i'll have to pass in `undefined` in those places. since they are all optional except for the job id, i think it would make sense to pass them in an object for more clarity.

Reply:
https://github.com/elastic/kibana/pull/70589 is inbound which changes this function to accept an object of options.

Reply:
Thank you, i'll change to using this, I had already created a similar API in the modules provider, but i'll remove that and use the latest one.